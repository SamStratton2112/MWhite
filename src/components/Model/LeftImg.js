import { Col, Row } from "react-bootstrap";
import List from "../List";
import './Model.css'

/** Used alongside RightImg to create a tiling effect alternating image and list.  */

const LeftImg = ({ image, list, name }) => {
    return (
        <Row className="center" >
            <Col className="center imgContainer">
                <img style={{ height: '30em' }} src={image} alt='Mike White Model' />
            </Col>
            <Col className="txtContainer">
                    <h2>{name}</h2>
                    <List list={list}/>
            </Col>
        </Row>
    )
};

export default LeftImg;