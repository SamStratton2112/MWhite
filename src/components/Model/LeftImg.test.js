import LeftImg from './LeftImg';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

const testList = ['test item 1', 'test item 2']

describe(LeftImg, () => {
    it("renders without crashing", () => {
        render(<MemoryRouter>
            <LeftImg name='testName' image="testImg" list={testList} />
        </MemoryRouter>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(<MemoryRouter>
            <LeftImg name='testName' image="testImg" list={testList}/>
        </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });
});