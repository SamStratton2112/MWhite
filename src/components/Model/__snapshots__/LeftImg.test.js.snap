// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftImg matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="center row"
  >
    <div
      class="center imgContainer col"
    >
      <img
        alt="Mike White Model"
        src="testImg"
        style="height: 30em;"
      />
    </div>
    <div
      class="txtContainer col"
    >
      <h2>
        testName
      </h2>
      <ul
        class="leftText scrollText"
        style="border-radius: .5em; color: black;"
      >
        <li>
          test item 1
        </li>
        <li>
          test item 2
        </li>
      </ul>
    </div>
  </div>
</DocumentFragment>
`;
